tests/test_node_util.py

```diff
@@ -114,3 +122,80 @@ def raise_node_dict(node_dict):
     raise_node_dict(node_dict)
     node_dict = {"node": [None]}
     raise_node_dict(node_dict)
+
+
+def test_invalid_project_graphs(simple_project_node, simple_material_node, simple_process_node, simple_property_node, simple_data_node, simple_computation_node, simple_computational_process_node):
+    project = copy.deepcopy(simple_project_node)
+    process = copy.deepcopy(simple_process_node)
+    material = copy.deepcopy(simple_material_node)
+
+    ingredient = cript.Ingredient(material=material, quantities=[cript.Quantity(key="mass", value=1.23, unit="gram")])
+    process.ingredients += [ingredient]
+
+    # Add the process to the experiment, but not in inventory or materials
+    # Invalid graph
+    project.collections[0].experiments[0].process += [process]
+    with pytest.raises(CRIPTOrphanedMaterialError):
+        project.validate()
+
+    # First fix add material to inventory
+    project.collections[0].inventories += [cript.Inventory("test_inventory", materials=[material])]
+    project.validate()
+    # Reverse this fix
+    project.collections[0].inventories = []
+    with pytest.raises(CRIPTOrphanedMaterialError):
+        project.validate()
+
+    # Fix by add to the materials list instead.
+    # Using the util helper function for this.
+    cript.add_orphaned_nodes_to_project(project, active_experiment=None)
```

Comment:
this could be a good idea, we don't implicitly fix it for them, but we give them a utility that they can fix it for themselves. 

I dont know, I think someone more senior than me would probably know the best design for this and how to go about it, it's a bit outside my depth of knowledge